src/factom-event-emitter.js

```diff
@@ -0,0 +1,282 @@
+const { EventEmitter } = require('events'),
+    Promise = require('bluebird'),
+    { isValidPublicFctAddress } = require('./addresses'),
+    { Transaction } = require('./transaction');
+
+const FACTOM_EVENT = {
+    directoryBlock: 'directoryBlock',
+    factoidBlock: 'factoidBlock',
+    adminBlock: 'adminBlock',
+    entryCreditBlock: 'entryCreditBlock',
+    newEntryChain: 'newEntryChain'
```

Comment:
My last question: why not just `newChain`? I don't think I have heard the term "entry chain" much before, because I think it is implicit that chains in Factom context are made of entries. `newEntryChain` could be confusing as one would wonder what kind of other chain there could be?